routing/pathfind.go

```diff
@@ -722,6 +722,13 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 		amountToSend := toNodeDist.netAmountReceived +
 			lnwire.MilliSatoshi(inboundFee)
 
+		// Check if accumulated fees would exceed fee limit when this
+		// node would be added to the path.
+		totalFee := int64(amountToSend) - int64(amt)
+		if totalFee > 0 && lnwire.MilliSatoshi(totalFee) > r.FeeLimit {
+			return
+		}
+
 		// Request the success probability for this edge.
 		edgeProbability := r.ProbabilitySource(
 			fromVertex, toNodeDist.node, amountToSend,
```

Comment:
I think we should also correct `weight := edgeWeight(netAmountToReceive, fee, timeLockDelta)`  to `weight := edgeWeight(amountToSend, fee, timeLockDelta)`

Reply:
in that case it should be `weight := edgeWeight(amountToSend, outboundFee, timeLockDelta)`?

Reply:
the fee is just passed in there to be converted to `int64`, but isn't really related to the time lock weight, which is a bit strange and can be addressed in a refactor, so I think `weight := edgeWeight(amountToSend, fee, timeLockDelta)` is correct